src/app/shared/issue-tables/issue-sorter.ts

```diff
@@ -0,0 +1,53 @@
+import { Issue, ISSUE_TYPE_ORDER, SEVERITY_ORDER } from '../../core/models/issue.model';
+import { MatSort } from '@angular/material';
+
+export function getSortedData(sort: MatSort, data: Issue[]): Issue[] {
+  if (!sort.active) {
+      return data;
+  }
+
+  const direction: number = (sort.direction === 'asc' ? 1 : -1);
+
+  return data.sort((a, b) => {
+    switch (sort.active) {
+      case 'type':
+        return direction * compareByIssueType(a.type, b.type);
+      case 'severity':
+        return direction * compareBySeverity(a.severity, b.severity);
+      case 'assignees':
+        return direction * compareByStringValue(a.assignees.join(', '), b.assignees.join(', '));
+      case 'teamAssigned':
+        return direction * compareByStringValue(a.teamAssigned.id, b.teamAssigned.id);
+      case 'Todo Remaining':
+        return -direction * compareByIntegerValue(a.numOfUnresolvedDisputes(), b.numOfUnresolvedDisputes());
+      case 'id':
+        return direction * compareByIntegerValue(a.id, b.id);
+      default: // title, responseTag are string values
+        return direction * compareByStringValue(a[sort.active], b[sort.active]);
+    }
+  });
+}
+
+function compareBySeverity(severityA: string, severityB: string): number {
+  const orderA = SEVERITY_ORDER[severityA];
+  const orderB = SEVERITY_ORDER[severityB];
+
+  return orderA < orderB ? -1 : 1;
```

Comment:
Slight nitpick - can we replace `orderA < orderB ? -1 : 1` with `compareByIntegerValue(orderA, orderB)` ?
Just to reduce duplication

(Same comment applies to `compareByIssueType` )

Reply:
ah yes, given that they are integers, we can use that function to help us deal with this!